firmware/Core/Src/littlefs/littlefs_helper.c

```diff
@@ -110,56 +108,80 @@ int8_t LFS_unmount()
 {
     if (!LFS_is_lfs_mounted)
     {
-        DEBUG_uart_print_str("LittleFS not mounted.\n");
+        LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_WARNING, LOG_all_sinks_except(LOG_SINK_FILE), "LittleFS not mounted!");
         return 1;
     }
 
     // Unmount LittleFS to release any resources used by LittleFS
     const int8_t unmount_result = lfs_unmount(&LFS_filesystem);
     if (unmount_result < 0)
     {
-        DEBUG_uart_print_str("Error un-mounting.\n");
+        LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_CRITICAL, LOG_all_sinks_except(LOG_SINK_FILE), "Error un-mounting LittleFS!");
         return unmount_result;
     }
 
-    DEBUG_uart_print_str("Successfully un-mounted LittleFS.\n");
+    LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_NORMAL, LOG_all_sinks_except(LOG_SINK_FILE), "LittleFS un-mounting successful!");
     LFS_is_lfs_mounted = 0;
     return 0;
 }
 
 /**
  * @brief Lists contents of LittleFS Directory
- * @param root_directory Pointer to cstring holding the root directory to open and read 
+ * @param root_directory Pointer to cstring holding the root directory to open and read
+ * @param offset Number of entries to skip before listing directory
+ * @param count Number of entries to list in total (if 0, prints all entries)
  * @retval 0 on success, 1 if LFS is unmounted, negative LFS error codes on failure
  */
-int8_t LFS_list_directory(const char root_directory[])
+int8_t LFS_list_directory(const char root_directory[], uint16_t offset, int16_t count)
 {
+    // Check if LFS is mounted
     if (!LFS_is_lfs_mounted)
     {
-        DEBUG_uart_print_str("LittleFS not mounted.\n");
+        LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_WARNING, LOG_all_sinks_except(LOG_SINK_FILE), "LittleFS not mounted!");
         return 1;
     }
 
+    // Try to open the directory
     lfs_dir_t dir;
     int8_t open_dir_result = lfs_dir_open(&LFS_filesystem, &dir, root_directory);
     if (open_dir_result < 0)
     {
-        DEBUG_uart_print_str("Error opening a directory.\n");
+        LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_CRITICAL, LOG_all_sinks_except(LOG_SINK_FILE), "Error opening directory: %s", root_directory);
         return open_dir_result;
     }
 
-    // result is positive on success, 0 at the end of directory, or  negative on failure.
+    if (count == 0) {
+        count = -1;
+    }
+
+    // result is positive on success, 0 at the end of directory, or negative on failure.
     int8_t read_dir_result = 1;
-    while (read_dir_result >= 0)
+    struct lfs_info info;
+    DEBUG_uart_print_str("Name \t bytes\n");
+    while (read_dir_result > 0)
```

Comment:
Regarding this while loop, there is a small thing that is occurring with the current code flow. if lfs_dir_read returns an error, you have to go through the entire loop before breaking and it could actually be weird behaviour when you try to print the information. 

Alternatives:
since count and read_dir_result both need to be valid, you could do something like:

```
for (read_dir_result = lfs_dir_read(...); read_dir_result > 0 && count != 0; count--) {
// check offset

// check info.type

}
```
You could also do a while loop which is just true, and breaks if lfs_dir_read is invalid

Lastly, you could also try a do-while loop if you really want.

Just some ideas, go with what you think makes the most sense.

Reply:
I didn't make any major changes to the loop structure, rather just moved the if statement